### PR TITLE
Peer reconnect on disconnect or error, take 2

### PR DESCRIPTION
**Pull request by carnesen**
Finally took the time to figure out what this whole rebasing thing is all about and am submitting the peer reconnect feature as this new pull request. Rebased to master, tests are passing on my laptop.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -7,13 +7,6 @@ if (require.main === module) {
   var config = require('config');
   var node = BitcoreNode.create(config.get('BitcoreNode'));
   node.start();
-  node.on('error', function(err) {
-    if (err.code === 'ECONNREFUSED') {
-      console.log('Connection to bitcoind failed');
-    } else {
-      console.log('Unrecognized error: ', err);
-    }
-  });
```
Comment (maraoz) [inline index.js]: why did you remove this? I see you added a 'error' handler in BitcoreNode, but events are generally consumed outside of the `EventEmitter` object instance

**Comment by carnesen** [comment]
Good to know, thanks! I moved the error hander back up to index. Is the 'reconnect' event pattern not so good then either?

**Comment by maraoz** [comment]
It could be if you wanted to expose the fact that the `NetworkMonitor` has reconnected, but from your code I see you're using it for internal signaling. I'd rather use a method call for that (instead of an event), and maybe fire a 'reconnect' event once it reconnects... but that's not even necessary

**Comment by carnesen** [comment]
Good idea. It's much cleaner that way, and agreed that there's no need for a separate 'reconnect'.

**Comment by carnesen** [comment]
This PR has been obsoleted by events.
